File: lib/locationable.dart
class Locationable {
  String name;
  String description;
  bool reserved;

  Locationable(this.name, this.description) : reserved = false;

  @override
  String toString() {
    return "$name ($description) [${reserved ? '❌ Reserved' : '🟢 Available'}]";
  }
}

File: src/Locationable.ts
export default class Locationable {
  name: string;
  description?: string;
  state: string;

  constructor(name: string, description?: string) {
    this.name = name;
    this.description = description || '';
    this.state = "Free";
  }

  setReserver() {
    return this.state = "Reserved";
  }
  
  setFree() {
    return this.state = "Free";
  };

  toString() {
    return `Name: ${this.name}, Description: ${this.description}, State: ${this.state}`;
  }
}
